core/ir/x86/decode_table.c

```diff
@@ -7857,6 +7867,18 @@ const instr_info_t vex_W_extensions[][2] = {
   }, { /* vex_W_ext 109 */
     {OP_vmovd, 0x660f7e10, "vmovd", Ed, xx, Vd_dq, xx, xx, mrm|vex, x, tevexwb[136][0]},
     {OP_vmovq, 0x660f7e50, "vmovq", Ey, xx, Vq_dq, xx, xx, mrm|vex, x, tevexwb[136][2]},
+  }, { /* vex_W_ext 110 */
```

Comment:
> I don't see the VEX_W_EXT entries for these? 

Sorry I got confused with the various tables. I have updated the encoder (`op_instr`) to point to `vex_W_extensions` and the decoder (`third_byte_38`) to point to to `e_vex_extensions`.

I ran some tests manually with binutils 2.36

This is the assembly with vex and evex encodings for the four instructions
```
$ cat vnni.s
vpdpbusd %zmm31, %zmm24, %zmm0
{vex} vpdpbusd %xmm0, %xmm1, %xmm2

vpdpbusds %zmm31, %zmm24, %zmm0
{vex} vpdpbusds %xmm0, %xmm1, %xmm2

vpdpwssd %zmm31, %zmm24, %zmm0
{vex} vpdpwssd %xmm0, %xmm1, %xmm2

vpdpwssds %zmm31, %zmm24, %zmm0
{vex} vpdpwssds %xmm0, %xmm1, %xmm2
```
This shows the opcodes after assembling the file above
```
$ as vnni.s
$ objdump -d a.out

a.out:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <.text>:
   0:   62 92 3d 40 50 c7       vpdpbusd %zmm31,%zmm24,%zmm0
   6:   c4 e2 71 50 d0          {vex} vpdpbusd %xmm0,%xmm1,%xmm2
   b:   62 92 3d 40 51 c7       vpdpbusds %zmm31,%zmm24,%zmm0
  11:   c4 e2 71 51 d0          {vex} vpdpbusds %xmm0,%xmm1,%xmm2
  16:   62 92 3d 40 52 c7       vpdpwssd %zmm31,%zmm24,%zmm0
  1c:   c4 e2 71 52 d0          {vex} vpdpwssd %xmm0,%xmm1,%xmm2
  21:   62 92 3d 40 53 c7       vpdpwssds %zmm31,%zmm24,%zmm0
  27:   c4 e2 71 53 d0          {vex} vpdpwssds %xmm0,%xmm1,%xmm2
```
When run through `drdisas` these show the right disassembly
```
$ clients/bin64/drdisas
62 92 3d 40 50 c7
c4 e2 71 50 d0
62 92 3d 40 51 c7
c4 e2 71 51 d0
62 92 3d 40 52 c7
c4 e2 71 52 d0
62 92 3d 40 53 c7
c4 e2 71 53 d0
 62 92 3d 40 50 c7    vpdpbusd zmm0 {k0}, zmm24, zmm31
 c4 e2 71 50 d0       vpdpbusd xmm2, xmm1, xmm0
 62 92 3d 40 51 c7    vpdpbusds zmm0 {k0}, zmm24, zmm31
 c4 e2 71 51 d0       vpdpbusds xmm2, xmm1, xmm0
 62 92 3d 40 52 c7    vpdpwssd zmm0 {k0}, zmm24, zmm31
 c4 e2 71 52 d0       vpdpwssd xmm2, xmm1, xmm0
 62 92 3d 40 53 c7    vpdpwssds zmm0 {k0}, zmm24, zmm31
 c4 e2 71 53 d0       vpdpwssds xmm2, xmm1, xmm0
```
> If the test passed: does it not contain these variants? 

I think the test is using EVEX by default. Is there a way to specify VEX encoding?

>  Actually why are there these separate entries: do we only want the other entries below and not these? The two opcode forms in the manual vary only by operand size which our var-size operands handle in a single entry?

I think there are three EVEX entries apart from the two VEX entries you listed.

Reply:
I have updated the encoder (`op_instr`) to point to `vex_W_extensions` so it might be generating VEX encoded opcodes. How can I check what exact bytes it is generating?

Reply:
Turning on VERBOSE in ir_x86.c should print out the disasm w/ bytes for everything: could search that.  Could also add a manual test inside ir_x86.c that synthesizes one of these opcodes and encodes it and checks the raw bytes.

Reply:
I had to add an API with two sources (no mask) to generate the vex encodings. I have added a few tests for this and also added a manual test that verifies the raw bytes.